### PR TITLE
SSD takeover now loads your active character slot

### PR DESCRIPTION
**Pull request by Ldip999**
## About The Pull Request

SSD takeover will now load your active character slot (meaning name, gender, hair, eyes, age, ethnicity, flavour text, basically everything except subrace (human/robot/vatgrown)) with a check to see if the character is already in the round (optional, and based on config value)
Also adds a currently missing confirmation window, with the job also listed during confirmation.

## Why It's Good For The Game

Round removal is bad and we and other branches of ss13 have been trying to remedy it for years, and TGMC offers a variety of colourful options, including respawning after 30 minutes (With a fancy valhalla to test out things while waiting, great idea by the way!), taking over a newly born vatgrown soldier, the larva queue, taking over minion xenos, or least commonly used, taking over an SSD marine. 
This PR aims to address this, by trying to remedy the inherent problems that come with playing an already estabilished character, that will feel alien and such. Already, the names are randomized (meaning that thankfully, you don't have to play as candice balls), but the visual appearance, gender, and the randomized name just feels alien and uncomfortable, so this PR aims to address this by (attempting to) load in your current character slot.
## Changelog
:cl:
qol: Taken over SSD mobs will now load your active character slot if possible
/:cl:

### CODE REVIEW

**Comment by Ragecrabbo** [comment]
theology of the matrix, www thescottsmithblog com, person becoming agent

**Comment by Wisemonster** [comment]
Isn't this to prevent a bunch of "John TGMC" duplicate bodies lying around?

**Comment by Ldip999** [comment]
Define "John TGMC"

**Comment by Notamaniac** [comment]
John TGMC dies, then takes over a SSD body. Another John TGMC is now in the round. That John TGMC dies, then takes over another SSD body. There are now 3 John TGMC's in the round.

(though i'm assuming that 'load your character slot if available' means that there's some check to avoid duplicate characters in the round)

**Comment by Ldip999** [comment]
> with a check to see if the character is already in the round (optional, and based on config value)
Depends on the server config. If it's set so, then we already could have 50 John TGMCs, limited only by the respawn timer. This will use the same logic and config flags for detecting and rejecting duplicate characters

**Comment by Ldip999** [comment]
It is currently set so that you can't have duplicates

**Review by Lumipharon — COMMENTED** [review]
I don't get what this is trying to resolve? When you take an ssd body it random names you already, does it not?

**Comment by Ldip999** [comment]
> I don't get what this is trying to resolve? When you take an ssd body it random names you already, does it not?

You are correct, it does give you a random name, however it still feels alien. That's what this PR aims to remedy. To make the taken over character more yours and that you'll feel more comfortable playing it, not to mention carrying over all RP aspects your character (that isn't already in the round) may have, such as friendships and such. Overall a better, more fun, more comfortable experience.

File: code/datums/actions/observer_action.dm [hunk 1]
```diff
@@ -85,13 +87,20 @@
...
 		to_chat(owner, span_warning("You are jobbaned from the [new_mob?.job.title] role."))
 		return
+
+	if(!ishuman(new_mob))
+		message_admins(span_adminnotice("[owner.key] took control of [new_mob.name] as [new_mob.p_they()] was ssd."))
+		log_admin("[owner.key] took control of [new_mob.name] as [new_mob.p_they()] was ssd.")
+		new_mob.transfer_mob(owner)
+		return
+	if(CONFIG_GET(flag/prevent_dupe_names) && GLOB.real_names_joined.Find(owner.client.prefs.real_name))
+		to_chat(usr, "<span class='warning'>Someone has already joined the round with this character name. Please pick another.<spawn>")
```
Comment (TiviPlus) [inline code/datums/actions/observer_action.dm:97]: use span class macros and fix wherever you pasted this from as well

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by Ldip999** [comment]
Like so?

**Comment by TiviPlus** [comment]
correct

**Pull request merged**